modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSView.tsx

```diff
@@ -109,7 +113,11 @@ const FDSView = ({
 			if (responseJSON?.id) {
 				setFDSView(responseJSON);
 
-				setLoading(false);
+				getFields(responseJSON).then((fields) => {
+					setTreeItems(fields);
+
+					setLoading(false);
+				});
```

Comment:
Can do both requests in a single one? Something like we are doing [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Settings.tsx#L55). `nestedFields` is a bit odd in above
```js
${API_URL.FDS_VIEWS}/${fdsViewId}?nestedFields=${OBJECT_RELATIONSHIP.FDS_ENTRY_FDS_VIEW}
```
It references parent object, where typically we reference children. I vaguely remember, is the purpose of this to load full FDSEntry object and not just a reference? :thinking: Even if we need it, would it work if we have a comma-separated list of nested fields, like [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/filters/Filters.tsx#L473), but we include entries and fields?

Reply:
We could try to do one request to retrieve all selected fields in the way you mention, instead of one per visualizationMode. It will require some refactors to extract each visualization mode fields and pass to the proper component.

`getFields` need the result of the previous Promise, so I cannot use `Promise.all` here

Reply:
Right, right, we need the endpoint of the view first. I'm not sure if it's possible to squash this into one request, probably not :thinking: With this in mind, current solution LGTM :+1: 

> We could try to do one request to retrieve all selected fields in the way you mention, instead of one per visualizationMode. It will require some refactors to extract each visualization mode fields and pass to the proper component

Aren't you doing this in this PR? We're passing `treeItems` into `Content`, that resolves into all tabs.

Reply:
In this PR we are getting all the fields (`openapi.json`) but we are making 3 requests for each of the visualization modes.

1 - `nestedFields`  -> FDSView details
2 - `openapi.json` -> Valid fields for the tree
3 - `fields?...` -> table visualization mode selected fields
4 - `list-sections?...` -> list visualization mode selected fields
5 - `cards-sections?...` -> cards visualization mode selected fields

![image](https://github.com/liferay-frontend/liferay-portal/assets/132653342/f25b34ab-0b92-4a7e-965a-94770d3d1804)

Reply:
Before this PR, every time we openened the field selection modal there were a request to `openapi.json`. With this PR, only one request when entering the View.

Reply:
Oh, I see what you mean. I think we can leave it as it is for now, this PR is already big enough. But it makes sense to me; if admin is changing table visualization mode, he is quite likely to change the other two. The hit to performance between fetching of single and multiple nested fields is very likely negligible.